lldb/source/Plugins/Language/CPlusPlus/LibStdcpp.cpp

```diff
@@ -287,8 +291,52 @@ bool lldb_private::formatters::LibStdcppStringSummaryProvider(
       } else
         return true;
     } break;
-    case eAddressTypeHost:
-      break;
+    case eAddressTypeHost: {
+
+      DataExtractor data;
+      Status error;
+      valobj.GetData(data, error);
+      if (error.Fail())
+        return false;
+
```

Comment:
Where is the code that detects the short string optimization? I don't see it here anywhere?

Reply:
Wouldn't short string optimization be covered by the check at the address type of children (Line 303)? If the children are host we directly read the std::string from the data extractor